docs/codebase-notes/sources-reducer.md

```diff
@@ -0,0 +1,251 @@
+
+### `"INSERT_SOURCE_ACTORS"`
+
+Just `return insertSourceActors(state, action);`
+
+Dispatched by: `actions/sources/newSources.js` line 254, in `newGeneratedSources()`
+
+### `"REMOVE_SOURCE_ACTORS"`
+
+Just `return removeSourceActors(state, action);`
+
+- **Dead**: dispatched in a `removeSourceActors` thunk, but that is not used anywhere
+
+### `"SET_SELECTED_LOCATION"`
+
+Creates a `location` value, possibly saves it as `prefs.pendingSelectedLocation` (BAD!), updates `state.selectedLocation` and `state.pendingSelectedLocation`.
+
+Dispatched by: `actions/sources/select.js` line 133
+
+### `"SET_VIEWPORT"`
+
+Relies on a check of `gToolbox.currentTool` (ew, globals in a reducer! This should probably be moved into a thunk)
+
+Dispatched by: `components/Editor/index.js` line 112
+
+Note: a couple other functions in the codebase named `updateViewport`, but they're completely different (`server/actors/` folder, related to UI layout)
+
+### `"CLEAR_SELECTED_LOCATION"`
+
+Creates a location object with an empty URL, saves it as `prefs.pendingSelectedLocation` (BAD!), and clears `state.selectedLocation`.
+
+Dispatched by: `actions/sources/select.js`, line 99 in `deselectSource()` and line 107 in `selectLocation()`
+
+### `"SET_PENDING_SELECTED_LOCATION"`
+
+Creates a location object with `{url, line, column}`, saves the pref (BAD!), and sets `state.pendingSelectedLocation`.
+
+Dispatched by: `actions/sources/select.js`, line 75 in `selectSourceURL`
+
+### `"LOAD_SOURCE_TEXT"`
+
+Just `return updateLoadedState(state, action);`
+
+Dispatched by: a `loadSourceTextPromise` function on line 53 in `actions/sources/loadSourceText.js`. This relies on the `redux-promise` middleware. `loadSourceTExtPromise` in turn is dispatched as part of `loadSourceText` on line 68, which is an instance of `memoizableAction`.
+
+### `"BLACKBOX"`
+
+Checks if `action.status === "done"`. If yes, tries to update a "blackboxed list" for that URL (related to prefs and tabs), and then updates a cache entry with `isBlackboxed`.
+
+### `"SET_ORIGINAL_BREAKABLE_LINES"`
+
+Updates `state.breakableLines[sourceId] = breakableLines`
+
+- **Dead** - not referenced anywhere
+
+### `"ADD_BREAKPOINT_POSITIONS"`
+
+Updates `state.breakpointPositions[sourceId]` and merges in new positions.
+
+Dispatched by: `actions/breakpoints/breakpointPositions.js` line 93, inside `_setBreakpointPositions`. This is another instance of `memoizableAction`
+
+### `"NAVIGATE"`
+
+Resets state, but increments `state.epoch`
+
+- **Dead**? Handled by a half-dozen reducers to reset their state, but does not appear to be dispatched anywhere
+
+### `"SET_FOCUSED_SOURCE_ITEM"`
+
+Just sets `state.focusedItem`.
+
+Dispatched by: `components/PrimaryPanes/SourcesTree.js`, line 134 when a sources tree item is focused
+
+### `"SOURCES_LOADED"`
+
+Just sets `state.sourcesLoading = false`.
+
+Dispatched by: `debugger/src/client/index.js` line 44, after waiting for `ThreadFront.findSources()` to complete
```

Comment:
These docs are great, but I worry that they might be hard to maintain. Would it make sense to start adding more inline documentation and use a docgen tool internally?

Fine to add these now and then come back later

Reply:
These aren't meant to be permanent. These are my scratch notes as I'm just trying to sort through what this codebase is doing in the first place. In a sense, them having value to anyone else is an incidental bonus.  

But, the info _is_ potentially useful in the near term, so it's worth committing these instead of me just keeping them locally on my own machine.